### ISSUE TITLE
Message list filter

### ISSUE THREAD

**Issue by cketti** [k9mail/k-9#4330]
In K-9 Mail 5.600 and earlier we supported only showing unread messages of a folder by clicking the unread count in the folder list. Same with starred messages. This functionality is missing in the current beta version (5.701).

But instead of recreating the old mechanism I'd like to add a filter button to the message list screen. For a first version we could use the following filter options:
- show unread messages
- show starred messages
- show messages with attachments

**Comment by clhunsen** [comment]
This is also missing for me. Although, I had a different idea here: How about adding "meta folders" to the _folder list_ to bring back this functionality? 

In more detail, the folder list for all account (including the unified inbox!) would contain the following folders:
- Unread messages
- Starred messages
- Messages with attachments
- some kind of separator to distinguish meta folders from "physical" folders
- the "physical" folders of the account (none here for the unified inbox, I guess)

This way, the feature would be placed more prominently and could be applied not only for the current folder but also for the current account (which resembles the "old way" more closely). What do you think?

**Comment by cketti** [comment]
@clhunsen: See #3398

**Comment by axelsimon** [comment]
Sorry to be that person but, "any progress on this?".

Thanks!

**Comment by Kyoshiro-san** [comment]
> But instead of recreating the old mechanism I'd like to add a filter button to the message list screen. For a first version we could use the following filter options:
> 
> * show unread messages
> * show starred messages
> * show messages with attachments

Thanks for taking this issue.
Does it mean it won't be cross-folder and only apply on current folder ?

**Comment by andrewfg** [comment]
My suggestion would be to show the respective filtered list when one touches the respective star/blob on the drawer - see below..

![image](https://user-images.githubusercontent.com/893994/134505130-72163119-8ac8-4d30-8810-7b800abf94c0.png)

**Comment by cketti** [comment]
@andrewfg: These are way too small to be touch targets. And not being recognizable as touch targets separate from the whole row makes it very confusing. An issue that popped up very often with the account overview screen in 5.600 was that users would accidentally click on the number of unread or starred messages and then complain that K-9 Mail was not showing all messages. They didn't realize they opened a filtered view.

A similar issue can be seen in Android's notification settings screen, even though there is a visual indicator separating the touch targets. A lot of people do not realize tapping the text part of a row and tapping the switch control do different things. See e.g. https://forum.k9mail.app/t/notification-sound/2021

<img src="https://user-images.githubusercontent.com/218061/134506733-9796cf3b-7987-45a8-965c-d7db89597269.png" alt="k9mail_notification_settings" width=300>

**Comment by Kyoshiro-san** [comment]
I agree, I myself sometimes used to misclick on the star or unread icon while I wanted to get to the account itself and now it's even smaller !

Although I'd be glad if we could find a way to restore previous functionnality, which was:
* access to all unread messages accross *all* folders from one account (not only currently displayed one).
* access to all starred messages accross *all* folders from one account (not only currently displayed one).

I'm emulating this by setting all folders as part of unified inbox, and sorting unified with unread emails first but it has a few issues: 
1/ it's multi account, and requires to add all folders to unified, killing the real purpose of unified inbox
2/ the < previous  / next > buttons don't go to the previous or next message in the displayed list
3/ it's refreshing slower than previous "see all unread messages" feature and causes misclicks, as the displayed message takes a bit of time to disappear after getting back to the list

**Comment by cketti** [comment]
This issue is just for filtering message list views (of any kind). To support use cases like the ones described the idea is to have a more powerful search (#4513). Maybe along with saved search (#3398).

**Comment by Kyoshiro-san** [comment]
I understand.
Would it be possible then to keep #5466 and #5467 opened ?

Thanks a lot !

**Comment by cketti** [comment]
> Would it be possible then to keep #5466 and #5467 opened ?

I don't see a point in keeping those issues open. They propose a solution that is not acceptable (as discussed in this issue).

**Comment by galapogos01** [comment]
Super keen for this feature to come back.

**Comment by pfolk** [comment]
What is needed to get this back in a milestone so it can actually be finished/implemented?
Would the feature as envisioned allow for showing only starred and unread messages at the top, as it currently can show unread at the top?
